src/image-copy/azext_imagecopy/custom.py

```diff
@@ -21,9 +21,17 @@ def imagecopy(cmd, source_resource_group_name, source_object_name, target_locati
               target_resource_group_name, temporary_resource_group_name='image-copy-rg',
               source_type='image', cleanup='false', parallel_degree=-1, tags=None, target_name=None,
               target_subscription=None, export_as_snapshot='false', timeout=3600):
+    # If --cleanup is set, forbid using an existing temporary resource group name.
+    # It is dangerous to clean up an existing resource group.
+    cli_cmd = prepare_cli_command(['group', 'exists', '-n', temporary_resource_group_name],
```

Comment:
Is there any existing function to do this check ? I'm not sure whether it is a good idea to run another cli command in current one.

Reply:
We can also use SDK. This extension runs cli commands instead of using SDK. I follow this style.

Reply:
okay... Personally, I do not suggest to follow this bad pattern.🤔

Reply:
@arrownj, 
This pattern was the **recommendation of the Azure CLI team** when this extension was developed. A refactor after a couple of years is always an option...

I know it might be difficult to have full context of the situation which is why I personally try to learn the history before voicing my opinion.